src/classes/UTIL_Describe.cls

```diff
@@ -865,4 +866,33 @@ public class UTIL_Describe {
 
         return recTypeId;
     }
+    /*******************************************************************************************************
+    * @description Cache for SObject type to List of ChildRelationshipNames
+    */
+    private static Map<SObjectType, List<String>> sObjectTypeToChildRelationshipNames = new Map<SObjectType, List<String>>();
+    
+
+    /*******************************************************************************************************
+    * @description Returns the list of all Child Relationship Name for a given Sobject Type
+    * @return sObjectTypeToChildRelationshipNames.get(sObjectTypeName) List of all the child Relationship Names
+    */
+    public static List<String> getAllChildRelationshipNames(SobjectType sObjectTypeName) {
+        if (sObjectTypeToChildRelationshipNames.get(sObjectTypeName) == NULL) {
+            
+            List<String> childRelationshipNames = new List<String>();
+            try {
+                DescribeSObjectResult describe = sObjectTypeName.getDescribe();
+                for (Schema.ChildRelationship childRelationship : describe.getChildRelationships()) {
+                    if (childRelationship != NULL && childRelationship.getRelationshipName() != NULL) {
+                        childRelationshipNames.add(childRelationship.getRelationshipName());
+                    }
+                }
+                sObjectTypeToChildRelationshipNames.put(sObjectTypeName, childRelationshipNames);
+             } catch (Exception ex) {
+                 return NULL;
+               }  
```

Comment:
@skosgi1 Please resolve the spacing issues on these lines.

Please have this return an empty string list. By forcing this to return null, we are creating behavior that we will need to account for further on in our code. If we can ensure that a method will never return null, then we can eliminate the handling of nulls for that method further on in our code.

If we want to have the behavior handled as though it is an error state, we should handle that as an exception so our code fails in a way that is informative to us and is meaningful to us when performing unit tests. We can catch the exception thrown and provide a custom exception with a custom label that is more informative to the end-user and debugger to the error that has occurred or we can preserve the underlying exception and test for that exception in our unit tests, failing the negative unit test if we encounter an unexpected exception.

I would prefer that we move away from any instance of try catch block that is generic enough to have the catch block be restated as catch{} as it can hide underlying unexpected behavior that can surface larger problems in the code.

Reply:
Thanks @MatthewBlanski I updated the code based on our discussion. Can you please give it another shot? Thx.